server/monitor-types/globalping.js

```diff
@@ -0,0 +1,530 @@
+    creditsHelpLink = "https://dash.globalping.io?view=add-credits";
+
+    supportsConditions = true;
+    conditionVariables = [
+        new ConditionVariable("record", defaultStringOperators ),
+    ];
+
+    /**
+     * @inheritdoc
+     */
+    constructor(agent) {
+        super();
+        this.agent = agent;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async check(monitor, heartbeat, _server) {
+        const apiKey = await Settings.get("globalpingApiToken");
+        this.hasAPIToken = !!apiKey;
+        const client = new Globalping({
+            auth: apiKey,
+            agent: this.agent,
+        });
+
+        switch (monitor.subtype ) {
+            case "ping":
+                await this.ping(client, monitor, heartbeat);
+                break;
+            case "http":
+                await this.http(client, monitor, heartbeat);
+                break;
+            case "dns":
+                await this.dns(client, monitor, heartbeat);
+                break;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async ping(client, monitor, heartbeat) {
+        const opts = {
+            type: "ping",
+            target: monitor.hostname,
+            inProgressUpdates: false,
+            limit: 1,
+            locations: [{ magic: monitor.location }],
+            measurementOptions: {
+                packets: monitor.ping_count,
+                protocol: monitor.protocol,
+            },
+        };
+
+        if (monitor.protocol === "TCP" && monitor.port) {
+            opts.measurementOptions.port = monitor.port;
+        }
+
+        if (monitor.ipFamily === "ipv4") {
+            opts.measurementOptions.ipVersion = IpVersion[4];
+        } else if (monitor.ipFamily === "ipv6") {
+            opts.measurementOptions.ipVersion = IpVersion[6];
+        }
+
+        log.debug("monitor", `Globalping create measurement: ${JSON.stringify(opts)}`);
+        let res = await client.createMeasurement(opts);
+
+        if (!res.ok) {
+            // retry
+            res = await client.createMeasurement(opts);
+            if (!res.ok) {
+                if (Globalping.isHttpStatus(429, res)) {
+                    throw new Error(`Failed to create measurement: ${this.formatTooManyRequestsError()}`);
+                }
+                throw new Error(
+                    `Failed to create measurement: ${this.formatApiError(res.data.error)}`
+                );
+            }
+        }
+
+        log.debug("monitor", `Globalping fetch measurement: ${res.data.id}`);
+        let measurement = await client.awaitMeasurement(res.data.id);
+
+        if (!measurement.ok) {
+            // retry
+            measurement = await client.awaitMeasurement(res.data.id);
+            if (!measurement.ok) {
+                throw new Error(
+                    `Failed to fetch measurement (${res.data.id}): ${this.formatApiError(measurement.data.error)}`
+                );
+            }
+        }
+
```

Comment:
We already have a retry mechanism in place. Lets use this instead and abort here directly, or?